hashintel/hash
# Move page route to `/@shortname/<pageEntityUuid>`

## 🌟 What is the purpose of this PR?




This PR moves pages from `/<accountId>/<pageEntityUuid>` to `/@shortname/<pageEntityUuid>`.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203363157432081/1203491213980897/f) _(internal)_

## 🔍 What does this change?




- adds a `getServerSideProps` function for pages to get the `pageEntityId` and the `pageWorkspace` on first render
- renames the `RouteAccountInfo` react context to `RouteWorkspaceInfo`

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



- The `pageEntityId` is re-calculated in the sidebar even though it has already been calculated in `getServerSideProps`

## 🛡 What tests cover this?



None. This PR was manually tested.

## ❓ How to test this?



1. Run the HASH App locally, and log in
3. Navigate to a page, and observe the shortname in the URL. Ensure the page is behaving as expected.

## 📹 Demo
